temporalio/client.py

```diff
@@ -0,0 +1,935 @@
+                later ones.
+            workflow_query_reject_condition: When to reject a query.
+        """
+        return Client(
+            await temporalio.workflow_service.WorkflowService.connect(target_url),
+            namespace=namespace,
+            identity=identity,
+            data_converter=data_converter,
+            interceptors=interceptors,
+            workflow_query_reject_condition=workflow_query_reject_condition,
+        )
+
+    def __init__(
+        self,
+        service: temporalio.workflow_service.WorkflowService,
+        *,
+        namespace: str = "default",
+        identity: str = f"{os.getpid()}@{socket.gethostname()}",
+        data_converter: temporalio.converter.DataConverter = temporalio.converter.default(),
+        interceptors: Iterable["Interceptor"] = [],
+        workflow_query_reject_condition: Optional[WorkflowQueryRejectCondition] = None,
+    ):
+        """Create a Temporal client from a workflow service.
+
+        See :py:meth:`connect` for details on the parameters.
+        """
+        self._service = service
+        self._namespace = namespace
+        self._identity = identity
+        self._data_converter = data_converter
+        self._interceptors = interceptors
+        self._workflow_query_reject_condition = workflow_query_reject_condition
+
+        # Iterate over interceptors in reverse building the impl
+        self._impl: OutboundInterceptor = _ClientImpl(self)
+        for interceptor in reversed(list(interceptors)):
+            self._impl = interceptor.intercept_client(self._impl)
+
+    @property
+    def service(self) -> temporalio.workflow_service.WorkflowService:
+        """Raw gRPC service for this client."""
+        return self._service
+
+    @property
+    def namespace(self) -> str:
+        """Namespace used in calls by this client."""
+        return self._namespace
+
+    @property
+    def identity(self) -> str:
+        """Identity used in calls by this client."""
+        return self._identity
+
+    @property
+    def data_converter(self) -> temporalio.converter.DataConverter:
+        """Data converter used by this client."""
+        return self._data_converter
+
+    async def start_workflow(
+        self,
+        workflow: str,
+        *args: Any,
+        id: str,
+        task_queue: str,
+        execution_timeout: Optional[timedelta] = None,
+        run_timeout: Optional[timedelta] = None,
+        task_timeout: Optional[timedelta] = None,
+        id_reuse_policy: WorkflowIDReusePolicy = WorkflowIDReusePolicy.ALLOW_DUPLICATE,
+        retry_policy: Optional[temporalio.common.RetryPolicy] = None,
+        cron_schedule: str = "",
+        memo: Optional[Mapping[str, Any]] = None,
+        search_attributes: Optional[Mapping[str, Any]] = None,
+        header: Optional[Mapping[str, Any]] = None,
+        start_signal: Optional[str] = None,
+        start_signal_args: Iterable[Any] = [],
+    ) -> "WorkflowHandle[Any]":
```

Comment:
I just saw this in the docs but haven't tried it (take a look at the bottom of the page): https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING

Reply:
Ah, I see. To confirm, are you saying I should add this import everywhere instead of using strings? I don't see a benefit really, but I can.

Reply:
It's just nicer than using strings IMHO

Reply:
I am going to add this

Reply:
Done